### ISSUE TITLE
create migration command before an existing one

### ISSUE THREAD

**Issue by lucasantoniooficial** [laravel/ideas#2127]
I have cases as often happens that it is to make models and migrations at the same time and I end up creating the migration in the wrong order, I would find it interesting to have some parameter so much to create models with migration and just migration from a Before.

For example:
`php artisan make: model Order -m`

`php artisan make: model User -m --before = Orders`

That way I wouldn't have to rewrite each model and migration again.

**Comment by fletch3555** [comment]
I don't like the before argument here.  It implies the model is somehow before something else.  Also, the logic to identify what migration file to create it before is quite complex.

First, how do you identify the specific file? `Orders` is not enough to identify the filename.  Sure, by default it would be called something like, `2020_01_07_172937_create_orders_table.php`, but there's not guarantee of that.  Maybe the file was renamed.  Maybe it was created manually.  The ONLY way I see this working is if you supply the full filename, or if Laravel completely rewrites how migrations work.

Second, and arguably more important, since the filenames are prefixed with the timestamp, how long before is "before"?  1 second? 1 hour? 1 day? What if it's 1 day, but there's another migration 4 hours before?  What if it's 1 second, and you add one "before", but then realize you need another one in between those two?

In short, given the complexity of this approach, compared to the simplicity of simply renaming the file manually (remember, it's not executed at the time of creation), I must vote against this suggestion

**Comment by marcosrocha85** [comment]
Whenever I have a similar problem like him (when some migration is miscreated or created on incorrect order), I just change the timestamp of file and everything works fine.

**Comment by lucasantoniooficial** [comment]
> Sempre que tenho um problema semelhante a ele (quando alguma migração é criada incorretamente ou criada em ordem incorreta), apenas altero o carimbo de data e hora do arquivo e tudo funciona bem.


I thought of something like changing the stamp of each file of the other so I would say that one created before the other.

**Comment by marcosrocha85** [comment]
> 
> 
> > Sempre que tenho um problema semelhante a ele (quando alguma migração é criada incorretamente ou criada em ordem incorreta), apenas altero o carimbo de data e hora do arquivo e tudo funciona bem.
> 
> I thought of something like changing the stamp of each file of the other so I would say that one created before the other.

If you take a look at [Migrator](https://github.com/laravel/framework/blob/951e10fbfed2a2d547c3fc33853125b4a619e01a/src/Illuminate/Database/Migrations/Migrator.php#L462) class, you will see that it parses directory and consequently runs migrations in created order.

**Comment by lucasantoniooficial** [comment]
> > > Sempre que tenho um problema semelhante a ele (quando alguma migração é criada incorretamente ou criada em ordem incorreta), apenas altero o carimbo de data e hora do arquivo e tudo funciona bem.
> > 
> > 
> > I thought of something like changing the stamp of each file of the other so I would say that one created before the other.
> 
> If you take a look at [Migrator](https://github.com/laravel/framework/blob/951e10fbfed2a2d547c3fc33853125b4a619e01a/src/Illuminate/Database/Migrations/Migrator.php#L462) class, you will see that it parses directory and consequently runs migrations in created order.


I thought of something like taking a class that doesn't inform before and changing or initial stamp with the new one.

**Comment by joelharkes** [comment]
Just change the timestamp on the migration to make it execute before the other.

Or put a 9999 year date stamp on it to always execute last :) after any newly created migrations, that's what we do.
